appservice/src/createAppService/createWebApp.ts

```diff
@@ -19,21 +21,30 @@ export async function createWebApp(
     return await createAppService(AppKind.app, actionContext, subscriptionContext, createOptions, showCreatingTreeItem);
 }
 
-export async function setWizardContextDefaults(wizardContext: IAppServiceWizardContext): Promise<void> {
-    await LocationListStep.setLocation(wizardContext, 'centralus');
-    // defaults that for if one workspace is opened
+export async function setWizardContextDefaults(wizardContext: IAppServiceWizardContext, actionContext: IActionContext, advancedCreation?: boolean): Promise<void> {
+    // only detect if one workspace is opened
     if (workspace.workspaceFolders && workspace.workspaceFolders.length === 1) {
-        await workspace.findFiles('package.json').then((files: Uri[]) => {
-            if (files.length > 0) {
-                wizardContext.newSiteOS = WebsiteOS.linux;
-            }
-        });
+        const fsPath: string = workspace.workspaceFolders[0].uri.fsPath;
+        if (await fse.pathExists(path.join(fsPath, 'package.json'))) {
+            wizardContext.recommendedSiteRuntime = LinuxRuntimes.node;
+        } else if (await fse.pathExists(path.join(fsPath, 'requirements.txt'))) {
+            // requirements.txt are used to pip install so a good way to determine it's a Python app
+            wizardContext.recommendedSiteRuntime = LinuxRuntimes.python;
+        }
+        actionContext.properties.recommendedSiteRuntime = wizardContext.recommendedSiteRuntime;
 
-        await workspace.findFiles('*.csproj').then((files: Uri[]) => {
-            if (files.length > 0) {
-                wizardContext.newSiteOS = WebsiteOS.windows;
+        if (!advancedCreation) {
+            await LocationListStep.setLocation(wizardContext, 'centralus');
+            // we only set the OS for the non-advanced creation scenario
+            // tslint:disable-next-line:strict-boolean-expressions
+            if (wizardContext.recommendedSiteRuntime) {
+                wizardContext.newSiteOS = WebsiteOS.linux;
             }
-        });
+            await workspace.findFiles('*.csproj').then((files: Uri[]) => {
```

Comment:
Wrap this in an else case. No need to search files if we already know it's linux